Food_Tracker_App/app/src/main/java/com/example/foodtracker/ui/ingredients/IngredientsMainScreen.java

```diff
@@ -160,4 +172,58 @@ private void initializeData() {
             adapter.notifyItemRangeInserted(0, ingredientArrayList.size());
         });
     }
+
+    /**
+     * Initializes the sort spinner
+     */
+
+    private void createSortSpinner(){
+        Spinner sortSpinner= findViewById(R.id.sort_spinner);
+        ArrayAdapter<CharSequence> sortAdapter = ArrayAdapter.createFromResource(this,
+                R.array.sortIngredients, android.R.layout.simple_spinner_item);
+        sortAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        sortSpinner.setAdapter(sortAdapter);
+
+        sortSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+
+                String selectedCol = parent.getItemAtPosition(position).toString();
+                sortIngredientQuery(selectedCol);
+
+            }
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+            }
+        });
+    }
+
+    /**
+     * Copyright: CC BY-SA 4.0.
+     * Resource: https://stackoverflow.com/questions/53474789/how-to-get-all-documents-from-a-firestore-collection-and-return-them-in-an-array
+     *
+     * Calls ingredientsCollection to do a query and returns the values from the query
+     * If task is successful, iterate through the query and add it ingredientArrayList
+     * then update adapter.
+     * @param sortCategory
+     */
+
+    private void sortIngredientQuery(String sortCategory){
+
+        ingredientsCollection.sortCollection(sortCategory, new OnCompleteListener<QuerySnapshot>() {
+            @Override
+            public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                if (task.isSuccessful()){
+                    ingredientArrayList.clear();
+                    for (QueryDocumentSnapshot document : task.getResult()){
+                        Ingredient newOrderIngredient = document.toObject(Ingredient.class);
+                        ingredientArrayList.add(newOrderIngredient);
```

Comment:
lines 218-220 iterates through the query, and for each document, we convert it into an Ingredient object, then add that into ingredientArrayList.

Reply:
Is there an alternative way to update this list, besides clearing the arrayList then adding with the ordered items?